deps/jerry/jerry-ext/handle-scope/handle-scope.c

```diff
@@ -0,0 +1,192 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <stdlib.h>
+#include "jerryscript-ext/handle-scope.h"
+
+const jerryx_handle_scope_t jerryx_handle_scope_root;
+jerryx_handle_scope jerryx_handle_scope_current = (jerryx_handle_scope) &jerryx_handle_scope_root;
+
+/**
+ * Opens a new handle scope and attach it to current global scope as a child scope.
+ *
+ * @return status code, jerryx_handle_scope_ok if success.
+ */
+jerryx_handle_scope_status
+jerryx_open_handle_scope(jerryx_handle_scope *result)
+{
+  jerryx_handle_scope scope = malloc(sizeof(jerryx_handle_scope_t));
```

Comment:
```
jerryx_handle_scope scope = malloc (sizeof (jerryx_handle_scope_t));
```

Reply:
I just have a performance concern on these `malloc`, and it creates somehow memory fragment, too. How about creating a configurable `jerryx_handle_scope` pool, and use one of them directly.

Reply:
in my opinion, you should not use handle scope if you invoke a function great frequency, so don't worry about it

Reply:
Hmm, once the napi always uses handle scope, so that's unacceptable if this leaves fragments in memory.